src/analytics/internal/constants.js

```diff
@@ -0,0 +1,173 @@
+export const API_HOST =
+    process.env.NODE_ENV === 'production'
+        ? 'https://2s1jj0js02.execute-api.eu-central-1.amazonaws.com/production'
+        : 'https://a8495szyaa.execute-api.eu-central-1.amazonaws.com/staging'
+
+// This is used to change the event type into integer to optimize the space
+export const EVENT_TYPES = {
+    successfulSearch: {
+        id: 1,
+        notifType: 'successfulSearch',
+    },
+    paginateSearch: {
+        id: 2,
+        notifType: 'successfulSearch',
+    },
+    unsuccessfulSearch: {
+        id: 3,
+        notifType: 'unsuccessfulSearch',
+    },
+    datepickerByDropdownStartDate: {
+        id: 4,
+        notifType: 'datepicker',
+    },
+    datepickerClearStartDate: {
+        id: 5,
+        notifType: 'datepicker',
+    },
+    datepickerByDropdownEndDate: {
+        id: 6,
+        notifType: 'datepicker',
+    },
+    datepickerClearEndDate: {
+        id: 7,
+        notifType: 'datepicker',
+    },
+    bookmarkFilter: {
+        id: 8,
+        notifType: 'bookmarkFilter',
+    },
+    tagFilter: {
+        id: 9,
+        notifType: 'tagFilter',
+    },
+    domainFilter: {
+        id: 10,
+        notifType: 'domainFilter',
+    },
+    addTag: {
+        id: 12,
+        notifType: 'tagging',
+    },
+    deleteTag: {
+        id: 13,
+        notifType: 'tagging',
+    },
+    deleteResult: {
+        id: 14,
+    },
+    createResultBookmark: {
+        id: 15,
+        notifType: 'bookmark',
+    },
+    removeResultBookmark: {
+        id: 16,
+        notifType: 'bookmark',
+    },
+    createBrowserBookmark: {
+        id: 17,
+        notifType: 'bookmark',
+    },
+    removeBrowserBookmark: {
+        id: 18,
+        notifType: 'bookmark',
+    },
+    createPopupBookmark: {
+        id: 19,
+        notifType: 'bookmark',
+    },
+    removePopupBookmark: {
+        id: 20,
+        notifType: 'bookmark',
+    },
+    addPopupTag: {
+        id: 21,
+        notifType: 'tagging',
+    },
+    deletePopupTag: {
+        id: 22,
+        notifType: 'tagging',
+    },
+    blacklistSite: {
+        id: 23,
+        notifType: 'blacklist',
+    },
+    blacklistDomain: {
+        id: 24,
+        notifType: 'blacklist',
+    },
+    removeBlacklistEntry: {
+        id: 25,
+        notifType: 'blacklist',
+    },
+    'changeTrackingPrefopt-out': {
```

Comment:
Probably want to rename this `changeTrackingPrefOptOut` or maybe more simply `optOutTrackingPref`?  Same applies for the opt-in one.

Reply:
The state changes in the privacy and the state is `opt-in` and `opt-out` and I just use the same thing and we can change.
https://github.com/WorldBrain/Memex/blob/develop/src/options/privacy/actions.js#L18

Reply:
For consistency with other types we can use `changeTrackingPrefOptOut` and `changeTrackingPrefOptIn`.

Reply:
Sounds good. That `privacy/actions` module is only setting the event name externally used in piwik; anything in-code should still follow the naming conventions (will eventually update the linters to make this easier; it's on my backlog).

Reply:
For now, I am just using the `newState` but we can use `newState` after updating `OptIn` or `OptOut`.
I can commit this thing or you can commit here. :slightly_smiling_face:

Reply:
Ah I see what you mean now. I think you accidentally linked the `develop` branch's version of that module here, but in your branch it needs to do more than just the Piwik call:
https://github.com/mukeshkharita/WebMemex/blob/feature/tracking_internal/src/options/privacy/actions.js#L19

I would just move that `value === 'y' || value === true` condition to ternary checks on both the old analytic's `name` field and the new one's `type` field. No need to keep that `newState` variable around